### PR TITLE
Bluebutton-1091 basic terraform part2

### PR DESCRIPTION
**Pull request by RickHawesUSDS**
*Why*
Building on Part 1 which created a stateful deployment, this pull request does the stateless deployment. 

*What*
- FHIR Server Autoscale Group
- FHIR Server IAM roles, profiles and security groups
- FHIR server load balancer 
- ETL Server EC2 Instance
- ETL Server IAM roles, profiles and security groups

*Tickets*
Fixes BLUEBUTTON-1161 Autoscale Groups
Fixes BLUEBUTTON-1162 ELB Groups

*Decisions Made*
- The FHIR server load balancer is now a V2 NLB. Network load balancer because we terminate TLS at the server. V2 to use the current stuff. 

*Work Still To Do*
- Using dummy values for the AMIs
- Using dummy values for ingress security groups
- A set of CloudWatch Alarms
- VPC peering setup

### CODE REVIEW

File: terraform/env/test/stateless/main.tf [hunk 1]
```diff
@@ -14,4 +14,7 @@ module "stateless" {
     env               = "test"
     tags              = {application="bfd", business="oeda", stack="test", Environment="test"}
   }  
+
+  fhir_ami            = var.fhir_ami
```
[the remark this answers is not in the document]
Comment (RickHawesUSDS) [reply terraform/env/test/stateless/main.tf]: Right. This is a temp solution.

File: terraform/modules/resources/asg/main.tf [hunk 2]
```diff
@@ -1,82 +1,95 @@
...
-##
-# Launch configuration
-##
-data "template_file" "user_data" {
-  template = "${file("${path.module}/templates/user_data.tpl")}"
+  ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    security_groups = [var.mgmt_config.remote_sg,var.mgmt_config.vpn_sg,var.mgmt_config.tool_sg]
```
[the remark this answers is not in the document]
Comment (RickHawesUSDS) [reply terraform/modules/resources/asg/main.tf]: Right. Copy error. Will remove this code.

**Comment by RickHawesUSDS** [comment]
Please review again.

File: terraform/modules/resources/asg/main.tf [hunk 3]
```diff
@@ -110,28 +138,19 @@ resource "aws_autoscaling_group" "main" {
...
-    key                 = "Environment"
-    value               = "${var.env}"
-    propagate_at_launch = true
-  }
-
-  tag {
-    key                 = "Function"
-    value               = "app-AppServer"
-    propagate_at_launch = true
+  dynamic "tag" {
```
Comment (jzulim) [inline terraform/modules/resources/asg/main.tf]: I like this method of dynamic tags. nice

File: terraform/modules/resources/lb/main.tf [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  availability_zone = var.env_config.azs[count.index]
+  filter {
+    name    = "tag:Layer"
+    values  = [var.layer] 
+  }
+}
+
+# LB 
+#
+# Use NLB to allow the EC2 instance to terminate the TLS connection
```
Comment (jzulim) [inline terraform/modules/resources/lb/main.tf]: Good catch here. CLB won't do the trick.

**Pull request merged**